### PR TITLE
Update logrus to v1.6.0, fixes #3223

### LINKED ISSUE #3223: Registry bad pointer on Windows from github.com/konsorten/go-windows-terminal-sequences.EnableVirtualTerminalProcessing
Hi, I got this error while trying to run the registry from a Windows machine:

```error
runtime: bad pointer in frame github.com/konsorten/go-windows-terminal-sequences.EnableVirtualTerminalProcessing at 0xc00060d630: 0x280
fatal error: invalid pointer found on stack
```

The chain of dependency failure is:
* Docker Registry vCurrent
   * github.com/sirupsen/logrus v1.4.2
      * github.com/konsorten/go-windows-terminal-sequences v1.0.1

This is a known issue, see: https://github.com/sirupsen/logrus/issues/1112

Solution: Update to [logrus v1.6.0](https://github.com/sirupsen/logrus/commits/v1.6.0), updates github.com/konsorten/go-windows-terminal-sequences to [v1.0.3](https://github.com/konsorten/go-windows-terminal-sequences/commits/v1.0.3) which contains the fix.

logrus has in a later (unreleased) commit removed the dependency on konsorten/go-windows-terminal-sequences completely.

### PR DESCRIPTION
**Pull request by AndreasHassing**
Fixes #3223 by bumping logrus to v1.6.0, which in turn bumps github.com/konsorten/go-windows-terminal-sequences to v1.0.3 wherein the fix to bad pointer is found.

A reference to v1.0.1 is still needed, since docker/metrics has that reference (not sure if it is using it, but I digress).

### CODE REVIEW

**Comment by thaJeztah** [comment]
Thanks! Change itself looks good, but I see you forgot the DCO sign off in your commit message; https://github.com/docker/distribution/blob/master/CONTRIBUTING.md#sign-your-work

If you can amend your commit, and force-push, that should update the PR and make CI go green (no need to open a new PR if you amend your commit)

**Comment by AndreasHassing** [comment]
Thanks @thaJeztah, signed off on the commit 😊.

**Review by thaJeztah — APPROVED** [review]
Thanks! LGTM (not a maintainer)

**Comment by thaJeztah** [comment]
@manishtomar @dmcgowan ptal

File: go.sum [hunk 1]
```diff
@@ -68,6 +68,8 @@ github.com/json-iterator/go v1.1.7/go.mod h1:KdQUCv79m/52Kvf8AW2vK1V8akMuk1QjK/u
 github.com/julienschmidt/httprouter v1.2.0/go.mod h1:SYymIcj16QtmaHHD7aYtjjsJG7VTCxuUUipMqKk8s4w=
 github.com/konsorten/go-windows-terminal-sequences v1.0.1 h1:mweAR1A6xJ3oS2pRaGiHgQ4OO8tzTaLawm8vnODuwDk=
```
Comment (AndreasHassing) [inline go.sum:69]: Used by [github.com/docker/go-metrics](https://github.com/docker/go-metrics), else I would've removed it 😊.

**Comment by manishtomar** [comment]
@AndreasHassing Thanks for the PR. Could you update logrus in https://github.com/docker/go-metrics/ to latest version?

**Comment by thaJeztah** [comment]
don't think go-metrics depends on logrus, unless it's through the prometheus client

**Comment by manishtomar** [comment]
Oh yes 🤦 . You are right. Looking at `go mod graph` it seems that both logrus versions are using `go-windows-terminal-sequences` package:
```
$ go mod graph | grep win
github.com/sirupsen/logrus@v1.6.0 github.com/konsorten/go-windows-terminal-sequences@v1.0.3
github.com/sirupsen/logrus@v1.2.0 github.com/konsorten/go-windows-terminal-sequences@v1.0.1
```

However, checking `go mod graph` in logrus doesn't reveal that package being used. So, I am curious where that package is coming from:
```
manish:~/src/logrus (master)
$ go mod graph
github.com/sirupsen/logrus github.com/davecgh/go-spew@v1.1.1
github.com/sirupsen/logrus github.com/pmezard/go-difflib@v1.0.0
github.com/sirupsen/logrus github.com/stretchr/testify@v1.2.2
github.com/sirupsen/logrus golang.org/x/sys@v0.0.0-20191026070338-33540a1f6037
```

**Comment by AndreasHassing** [comment]
@manishtomar: the dependency is removed entirely from logrus between 1.6.0 and master, but they haven't released yet 😊.

**Comment by manishtomar** [comment]
@AndreasHassing Aah ok. Thanks a lot for the info.

**Pull request merged**